### PR TITLE
Display chart data in relative numbers

### PR DESCRIPTION
**Pull request by MicheleK17**
Ticket: #288
Change-Id: Ie8d9e6e1381211cef155900eed625574661fd9c0

### CODE REVIEW

File: src/app/Main/Body/QueryPage/Results/Charts/BarChartResults.component.tsx [hunk 1]
```diff
@@ -9,6 +9,7 @@ export function BarChartResultsComponent(props: {
     getPngDownloadUriRef: MutableRefObject<(() => Promise<string>) | null>;
     xAxisLabel: string;
     yAxisLabel: string;
+    xAxisMax: number | undefined;
```
Comment (GerberI) [inline src/app/Main/Body/QueryPage/Results/Charts/BarChartResults.component.tsx:12]: I recommend to use the short form `xAxisMax?: number` for optional props.

File: src/app/Main/Body/QueryPage/Results/Charts/QueryPageContent-BarChart.component.tsx [hunk 2]
```diff
@@ -108,14 +119,21 @@ export function QueryPageContentBarChartResultsComponent(props: {
...
-    const colName = props.chartData.column
-    const rowName = props.chartData.row
+    const colName = props.chartData.column;
+    const rowName = props.chartData.row;
 
-    const isChart = colName !== "" || rowName !== ""
+    const isChart = colName !== "" || rowName !== "";
+
+    let chartData = props.chartData.statisticDataAbsolute;
+    let xAxisMax;
```
Comment (GerberI) [inline src/app/Main/Body/QueryPage/Results/Charts/QueryPageContent-BarChart.component.tsx:128]: Recommendation: Use `let xAxisMax = undefined` to make it clear that it is not accidentally left undefined in the next part of the code.

File: src/app/Main/Body/QueryPage/Results/Charts/QueryPageContent-BarChart.component.tsx [hunk 3]
```diff
@@ -69,14 +72,22 @@ function processingTableDataToApexData(
...
+        xAxisLabel = `${t("Results.NrIsolatesText")} ${t("Results.per")} ${t(
             `Filters.${rowName}`
         )}`;
+        if (displayOption === "relative") {
+            xAxisLabel = `${t("Results.NrIsolatesTextRelative")} ${t("Results.per")} ${t(
+                `Filters.${rowName}`
+            )} ${t("Results.Unit")}`;
+        }
+    } else if (displayOption === "relative") {
+        xAxisLabel = `${t("Results.NrIsolatesTextRelative")} ${t("Results.Unit")}`;
```
Comment (GerberI) [inline src/app/Main/Body/QueryPage/Results/Charts/QueryPageContent-BarChart.component.tsx:90]: As the x axis label is now set differently in all if branches i suggest to reflect that in the code:
```
let xAxisLabel = "";
if(rowName !== "") {
  if(absolute) {...}
  else if(relative) {...}
} else {
  if(absolute) {...}
  else if(relative) {...}
}
```

File: src/locales/en/QueryPage.json [hunk 4]
```diff
@@ -7,7 +7,7 @@
     "Results": {
         "Parameter": "Selected Filters",
         "TableHead": "Number of isolates",
-        "TableHeadRelative": "Isolates in %",
+        "TableHeadRelative": "Isolates",
```
Comment (GerberI) [inline src/locales/en/QueryPage.json:10]: Reject: You forgot to rename TableHead and TableHeadRelative here as in the german file.

**Review by GerberI — CHANGES_REQUESTED** [review]

File: src/app/Main/Body/QueryPage/Results/Charts/QueryPageContent-BarChart.component.tsx (hunk 2 above)
Comment (MicheleK17) [reply src/app/Main/Body/QueryPage/Results/Charts/QueryPageContent-BarChart.component.tsx:128]: In this case the linter is complaining ("Do not use useless `undefined`.") I leave it as it is.

File: src/app/Main/Body/QueryPage/Results/Charts/BarChartResults.component.tsx (hunk 1 above)
Comment (MicheleK17) [reply src/app/Main/Body/QueryPage/Results/Charts/BarChartResults.component.tsx]: Done.

File: src/app/Main/Body/QueryPage/Results/Charts/QueryPageContent-BarChart.component.tsx (hunk 3 above)
Comment (MicheleK17) [reply src/app/Main/Body/QueryPage/Results/Charts/QueryPageContent-BarChart.component.tsx]: Done.

File: src/locales/en/QueryPage.json (hunk 4 above)
Comment (MicheleK17) [reply src/locales/en/QueryPage.json]: Done. Thx!

**Review by GerberI — APPROVED** [review]
Commit: Little changes
thx

File: src/app/Main/Body/QueryPage/Results/Charts/QueryPageContent-BarChart.component.tsx (hunk 2 above)
Comment (GerberI) [reply src/app/Main/Body/QueryPage/Results/Charts/QueryPageContent-BarChart.component.tsx:133]: lol

**Comment by GerberI** [comment]
All done, can be merged

**Pull request merged**